interop-tests/src/bin/wasm_ping.rs

```diff
@@ -0,0 +1,211 @@
+}
+
+async fn run_test(proxy_addr: &str, timeout: Duration) -> Result<()> {
+    // start a webdriver process
+    // currently only the chromedriver is supported as firefox doesn't
+    // have support yet for the certhashes
+    let mut driver = tokio::process::Command::new("chromedriver")
+        .arg("--port=45782")
+        .stdout(Stdio::piped())
+        .kill_on_drop(true)
+        .spawn()?;
+    // read driver's stdout
+    let driver_out = driver
+        .stdout
+        .take()
+        .context("No stdout found for webdriver")?;
+    // wait for the 'ready' message
+    let mut reader = BufReader::new(driver_out).lines();
+    while let Some(line) = reader.next_line().await? {
+        if line.contains("ChromeDriver was started successfully.") {
+            break;
+        }
+    }
+
+    // run a webdriver client
+    let mut caps = DesiredCapabilities::chrome();
+    caps.set_headless()?;
+    let driver = WebDriver::new("http://localhost:45782", caps).await?;
+    // go to the wasm test service
+    driver.goto(format!("http://{}", proxy_addr)).await?;
+    // wait for the script to finish and set the result
+    match driver
+        .query(By::Id("result"))
+        .wait(timeout, Duration::from_millis(200))
+        .first()
+        .await
+    {
+        // print the result
+        Ok(span) => {
+            println!("{}", span.text().await?);
+            driver.quit().await?;
+            Ok(())
+        }
+        // or return a timeout error
+        Err(e) => {
+            driver.quit().await?;
+            Err(e).context("Timed out waiting for the test result")
+        }
+    }
+}
+
+/// Redis proxy handler.
+/// `blpop` is currently the only redis client method used in a ping dialer.
+async fn redis_blpop(
+    State(client): State<Client>,
+    request: Json<BlpopRequest>,
+) -> Result<Json<Vec<String>>, StatusCode> {
+    let mut conn = client.get_async_connection().await.map_err(|e| {
+        warn!("Failed to connect to redis: {e}");
+        StatusCode::INTERNAL_SERVER_ERROR
+    })?;
+    let res = conn
+        .blpop(&request.key, request.timeout as usize)
+        .await
+        .map_err(|e| {
+            warn!(
+                "Failed to get list elem {} within timeout {}: {e}",
+                request.key, request.timeout
+            );
+            StatusCode::INTERNAL_SERVER_ERROR
+        })?;
+
+    Ok(Json(res))
+}
+
+/// Serve the main page which loads our javascript
+async fn serve_index_html() -> Result<impl IntoResponse, StatusCode> {
+    let html = r#"
+        <!DOCTYPE html>
+        <html>
+        <head>
+            <meta charset="UTF-8" />
+            <title>libp2p ping test</title>
+            <script type="module" src="./index.js"></script>
+        </head>
+
+        <body></body>
+        </html>
+    "#;
+
+    Ok(Html(html))
+}
+
+/// Serve a js script which runs the main test function
+async fn serve_index_js(
+    config: &config::Config,
```

Comment:
You could destruct this variable in the signature already and then directly interpolate them in the `format!` string:

```suggestion
    config::Config { transport, ip, is_dialer, test_timeout }: &config::Config,
```

Reply:
I'm not personally a fan of a longer destructures directly in function signatures as I find it less readable. destructured in fn body, hope that's okay